tests/test_url.py

```diff
@@ -649,8 +649,13 @@ def test_with_query():
 
 def test_with_query_kwargs():
     url = URL('http://example.com')
-    assert str(url.with_query(
-        {'a': '2', 'b': '4'}, a='1')) == 'http://example.com/?a=1'
+    with pytest.raises(ValueError):
+        url.with_query(
+            {'a': '2', 'b': '4'}, a='1')
+    with pytest.raises(TypeError):
+        url.with_query(b=3)
+    with pytest.raises(ValueError):
+        url.with_query('a=1', 'a=b')
     assert 'query2=1' in str(url.with_query(query='1', query2='1'))
```

Comment:
Hmm. Returning to hash randomization: you can check not `str(url)` but `url.query == {query='1', query2='1'}`.
It should work.
Please try this way.